src/main/java/org/opensearch/knn/index/query/BaseQueryFactory.java

```diff
@@ -63,6 +63,10 @@ public Optional<QueryShardContext> getContext() {
         public Optional<RescoreContext> getRescoreContext() {
             return Optional.ofNullable(rescoreContext);
         }
+
+        public Optional<Boolean> getExpandNested() {
+            return Optional.ofNullable(expandNested);
+        }
```

Comment:
should we add the default value directly here rather than in `src/main/java/org/opensearch/knn/index/query/KNNQueryFactory.java` ?

Reply:
I will let the factory to decide on default value to be consist with other values. For example, rescoreContext's default value is null but we are still wrapping it as Optional instead of just returning null.